### PR TITLE
Java Docs & kafka-test module

### PR DESCRIPTION
**Pull request by big-andy-coates**
- Add java docs (requirement for publishing to Maven Central)
- Move `SchemaRegistryContainer` into new `kafka-test` module.

### CODE REVIEW

File: parser/src/main/java/io/specmesh/apiparser/model/SchemaInfo.java [hunk 1]
```diff
@@ -14,6 +17,20 @@ public class SchemaInfo {
     private final String contentType;
     private String schemaLookupStrategy;
 
+    /**
+     * @param schemaRef
+     *            location of schema
+     * @param schemaFormat
+     *            format of schema
+     * @param schemaIdLocation
+     *            ???
```
Comment (bluemonk3y) [inline parser/src/main/java/io/specmesh/apiparser/model/SchemaInfo.java:26]: part of the async api spec - can be header or payload

File: parser/src/main/java/io/specmesh/apiparser/model/SchemaInfo.java [hunk 2]
```diff
@@ -14,6 +17,20 @@ public class SchemaInfo {
...
+     * @param schemaIdPayloadLocation
+     *            ???
```
Comment (bluemonk3y) [inline parser/src/main/java/io/specmesh/apiparser/model/SchemaInfo.java:28]: similar to above - a confluent serde assumes it - examples here: https://gist.github.com/dalelane/3931c17b14c51fa4a1cf25496237d188

**Review by bluemonk3y — APPROVED** [review]
left a comment about the schemaId location stuff - but probably redundant because different serdes will assume schemaid location and characteristics... i.e. it is already hard coded and likely to be ignored

File: parser/src/main/java/io/specmesh/apiparser/model/SchemaInfo.java (hunk 1 above)
Comment (big-andy-coates) [reply parser/src/main/java/io/specmesh/apiparser/model/SchemaInfo.java:26]: As agreed, you'll fix forward on these two.

**Pull request merged**